2_process/src/munge_inst_timeseries.R

```diff
@@ -55,3 +55,48 @@ aggregate_data_to_hourly <- function(inst_data,output_tz){
 
 }
 
+
+
+aggregate_data_to_daily <- function(inst_data, daily_data, min_daily_coverage, output_tz){
+  #' 
+  #' @description Function to aggregate instantaneous NWIS data collected at sub-hourly (e.g. 15/5/30 min) intervals to hourly min/mean/maxs
+  #'
+  #' @param inst_data a data frame containing the downloaded time series for NWIS instantaneous site. 
+  #' inst_data must include the following columns: c("Value_Inst","dateTime","site_no")
+  #' @param daily_data a data frame the downloaded daily time series of DO data. This is used so that if a site is already in 
+  #'  the daily sites, we won't do the aggregating here
+  #' @param min_daily_coverage - float 0-1, minimum coverage needed to return daily summaries.
+  #' A min_daily_coverage of 0.5 means that summary stats will be calculated for 
+  #' days with at least 50% coverage. Coverage is calculated on a daily basis
+  #'  as (num_non_na_vals/(num_non_na_vals + num_na_vals))
+  #' 
+  #' @param output_tz character string to set display attribute of dateTime. Possible values to provide are "UTC",
+  #' "America/New_York","America/Chicago", "America/Denver","America/Los_Angeles", "America/Anchorage", 
+  #' as well as the following which do not use daylight savings time: "America/Honolulu", "America/Jamaica",
+  #' "America/Managua","America/Phoenix", and "America/Metlakatla"
+  #'
+  #' @value A data frame containing daily min, mean, and max values the parameter of interest
+  
+  
+  only_inst_data = setdiff(inst_data$site_no, daily_data$site_no)
+  
+  daily_values <- inst_data %>%
+    filter(site_no %in% only_inst_data) %>%
+    mutate(dateTime_local = lubridate::with_tz(dateTime,tzone=output_tz),
+           Date = lubridate::date(dateTime_local)) %>%
+    group_by(Date, site_no, agency_cd, Parameter) %>%
```

Comment:
I noticed that the table that gets returned (e.g. by `tail(p2_inst_data_daily)`) looks a bit odd since it is sorted first by Date, and then by site number. The values should not change, but if we wanted to aesthetically match the p1_daily_data I think you should just be able to swap the order of `Date` and `site_no` here.

Reply:
good idea
```suggestion
    group_by(site_no, Date, agency_cd, Parameter) %>%
```